src/chain.h

```diff
@@ -180,14 +180,27 @@ class CBlockIndex
     uint32_t nBits{0};
     uint32_t nNonce{0};
 
+    /* YespowerSugar */
+    //! (currently memory only, but don't have to be)
+    bool cache_init{false};
+    uint256 cache_block_hash{};
+    uint256 cache_PoW_hash{};
+
     //! (memory only) Sequential id assigned to distinguish order in which blocks are received.
     int32_t nSequenceId{0};
 
     //! (memory only) Maximum nTime in the chain up to and including this block.
     unsigned int nTimeMax{0};
 
+    /* YespowerSugar */
+    void SetNullCacheInit()
```

Comment:
I am little bit confused applying PoW cache part in v0.21 which does not use SetNull(), however we may do in somewhere...